pkg/cluster/install.go

```diff
@@ -56,8 +56,15 @@ func (m *manager) AdminUpdate(ctx context.Context) error {
 
 func (m *manager) Update(ctx context.Context) error {
 	steps := []steps.Step{
-		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		steps.Action(m.initializeKubernetesClients), // must be first
 		steps.Action(m.initializeClusterSPClients),
+		steps.Action(m.clusterSPObjectID),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
```

Comment:
In general, it is better if this is first, because all the other steps can run in the knowledge that the dynamic validator passed.  Put it back to the top in a follow-up?